hdx/data/dataset.py

```diff
@@ -507,7 +509,7 @@ def search_in_hdx(query: str, include_gallery: Optional[bool] = True, **kwargs)
         dataset = Dataset()
         total_rows = kwargs.get('rows', sys.maxsize)
```

Comment:
Skeptical about this line, do you really want a generator going all the way up to sys.maxsize?